iree/tools/init_iree_passes.h

```diff
@@ -46,6 +47,7 @@ inline void registerAllIreePasses() {
 
   linalg_ext::registerLinalgExtPasses();
   IREE::Flow::registerFlowPasses();
+  IREE::Flow::registerTensorToFlowPassPipelines();
```

Comment:
@benvanik I was wondering if the new pass should be registered in `registerFlowPasses` instead of adding a new function. WDYT?

Reply:
ah, we don't want a pass for this here at all - we just want patterns that can be added to other passes
see for example:
https://github.com/google/iree/blob/552d3f888e46f96c8227b1bd2e4600b85338237b/iree/compiler/Dialect/HAL/Conversion/FlowToHAL/ConvertFlowToHAL.h
https://github.com/google/iree/blob/552d3f888e46f96c8227b1bd2e4600b85338237b/iree/compiler/Dialect/VM/Conversion/MemRefToVM/ConvertMemRefToVM.h

We want to add these patterns to the existing https://github.com/google/iree/blob/323108e3869648a42dbfc82f8eb54cf05bdbafef/iree/compiler/Dialect/Flow/Transforms/ConvertToFlowTensorOps.cpp (you'll note it already ended up growing some tensor->flow patterns that should be moved to your new Conversions/TensorToFlow/ directory)

so: delete all the pass stuff, replace with a populate patterns method, call that method from ConvertToFlowTensorOps

Reply:
Gotcha. That's the kind of feedback I was hoping for, since there's a mix of different styles depending on when compiler code was written or by whom >_>

I hesitated to only provide patterns, since there's quite a bit of boilerplate in the existing pass that I'm moving: https://github.com/google/iree/blob/1cd9f0ab6995d0333f7d76412a4bfdcc3c5d35b6/iree/compiler/InputConversion/Common/ConvertUpstreamToIREE.cpp#L135-L157

Reply:
The FlowToHAL has the setupFlowToHALLegality method that does those ConversionTarget manipulations - you can do the same here so that the tensor CastOp/FromElements/etc can be made illegal as needed.

Reply:
Still poking at this, but I think I'm moving in the right direction.

I feel like there should be a way to not have the conversion patterns and "regular" patterns separate. (might be not understanding the terminology + MLIR infra, or maybe some of the patterns should be converted to one or the other)

Reply:
Made those changes, but still working out a few test failures (since this now applies more patterns in several places).